File: python-examples/browser-sdk-showcase/api/download-file.py
# https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/download_file
from playwright.async_api import Page
from typing import TypedDict
from intuned_browser import download_file
# from intuned_runtime import extend_payload


class Params(TypedDict):
    timeout: int | None


async def automation(page: Page, params: Params | None = None, **_kwargs):
    timeout = params.get("timeout") if params else 15
    # Direct download from URL
    download = await download_file(page, 'https://intuned-docs-public-images.s3.amazonaws.com/32UP83A_ENG_US.pdf',timeout_s=timeout)
    print("Result of downloading file from URL:")
    print(download.suggested_filename)
    
    # Download from trigger
    await page.goto("https://sandbox.intuned.dev/pdfs")
    download = await download_file(page, page.locator("xpath=//tbody/tr[1]//*[name()='svg']"))
    print("Result of downloading file from trigger:")
    print(download.suggested_filename)
    
    # Download from trigger with custom callable function
    await page.goto("https://sandbox.intuned.dev/pdfs")
    download = await download_file(page, lambda page: page.locator("xpath=//tbody/tr[1]//*[name()='svg']").click())
    print("Result of downloading file from trigger with custom callable function:")
    print(download.suggested_filename)
    
    return download.suggested_filename

File: typescript-examples/browser-sdk-showcase/api/download-file.ts
// https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/downloadFile
import { BrowserContext, Page } from "playwright";
import { downloadFile } from "@intuned/browser";

interface Params {
  timeout?: number;
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  const timeout = params.timeout || 15000;

  // Direct download from URL
  let download = await downloadFile({
    page,
    trigger: "https://intuned-docs-public-images.s3.amazonaws.com/32UP83A_ENG_US.pdf",
    timeoutInMs: timeout,
  });
  console.log("Result of downloading file from URL:");
  console.log(download.suggestedFilename());

  // Download from trigger
  await page.goto("https://sandbox.intuned.dev/pdfs");
  download = await downloadFile({
    page,
    trigger: page.locator("xpath=//tbody/tr[1]//*[name()='svg']"),
  });
  console.log("Result of downloading file from trigger:");
  console.log(download.suggestedFilename());

  // Download from trigger with custom callable function
  await page.goto("https://sandbox.intuned.dev/pdfs");
  download = await downloadFile({
    page,
    trigger: async (page: Page) => {
      await page.locator("xpath=//tbody/tr[1]//*[name()='svg']").click();
    },
  });
  console.log("Result of downloading file from trigger with custom callable function:");
  console.log(download.suggestedFilename());

  return download.suggestedFilename();
}

